src/SkipOnEmptyInterface.php

```diff
@@ -4,9 +4,96 @@
 
 namespace Yiisoft\Validator;
 
+use Yiisoft\Validator\EmptyCriteria\NeverEmpty;
+use Yiisoft\Validator\EmptyCriteria\WhenEmpty;
+use Yiisoft\Validator\EmptyCriteria\WhenNull;
+use Yiisoft\Validator\Rule\Trait\SkipOnEmptyTrait;
+
+/**
+ * An optional interface for rules to implement. It allows skipping validation for a rule when the validated value is
+ * "empty". Requires an additional private class property `$skipOnEmpty`. In package rules it's `null` by default:
+ *
+ * ```php
+ * public function __construct(
+ *     // ...
+ *     private mixed $skipOnEmpty = null,
+ *     // ...
+ * ) {
+ * }
+ * ```
+ *
+ * The package ships with {@see SkipOnEmptyTrait} which already implements that interface. All you have to do is include
+ * it in the rule class.
+ */
 interface SkipOnEmptyInterface
 {
+    /**
+     * A setter to change `$skipOnEmpty` property. Must be implemented following immutability concept:
```

Comment:
Interface knows nothing about how the value is stored so we should not mention implementation details.

Reply:
Immuntability could be mentioned as a requirement, but I don't think an example is needed.